src/cozy/services/installation-guard.service.ts

```diff
@@ -0,0 +1,67 @@
+import { Injectable } from '@angular/core';
+import {
+    ActivatedRouteSnapshot,
+    CanActivate,
+    RouterStateSnapshot,
+} from '@angular/router';
+import { MessagingService } from '../../../jslib/src/abstractions/messaging.service';
+import { CozyClientService, IsInstalled } from './cozy-client.service';
+
+@Injectable({ providedIn: 'root' })
+export class VaultInstallationService {
+    constructor(private clientService: CozyClientService) {}
+
+    async IsVaultInstalled(): Promise<boolean> {
+        return IsInstalled.value;
+    }
+
+    setIsInstalled() {
+        IsInstalled.value = true;
+    }
+}
+
+@Injectable({ providedIn: 'root' })
+export class VaultInstalledGuardService implements CanActivate {
+    constructor(
+        private vaultInstallationService: VaultInstallationService,
+        private messagingService: MessagingService
+    ) {}
+
+    async canActivate(
+        route: ActivatedRouteSnapshot,
+        routerState: RouterStateSnapshot
+    ) {
+        const isInstalled =
+            await this.vaultInstallationService.IsVaultInstalled();
+
+        if (!isInstalled) {
+            this.messagingService.send('installBlocked');

```

Comment:
Does this event name already exist? Or do you created it?

Reply:
had my answer in the next commit: already exist

Reply:
Hmm nope, I created them in this PR so they did not exist before.
Those are events calleds by router guards. But they are badly named. Maybe `installationPageBlocked` and `redirectToInstallPage` ?

Reply:
Oh 🤕 . I saw an addition in the next commit, but forgot that you added these stuffs before...

The naming seems to be coherent with what we already have to me.